forest_benchmarking/rpe.py

```diff
@@ -1,227 +1,491 @@
+        1) Prepare the equal superposition between computational basis states (i.e. the
+            eigenvectors of a rotation about the Z axis)
+        2) Perform a change of basis which maps the computational basis to eigenvectors of the
+            rotation.
+        3) Perform the rotation depth-many times, where depth=2^iteration number. Each
+            eigenvector component picks up a phase from the rotation. In the 1-qubit case this
+            means that the state rotates about the axis formed by the eigenvectors at a rate
+            which is given by the relative phase between the two eigenvector components.
+        4) Invert the change of basis to return to the computational basis.
+        5) Prepare (one of) the qubit(s) for measurement along either the X or Y axis.
+        6) Measure this qubit, and in the multi-qubit case other qubits participating in rotation.
+    Measure_qubits can be used e.g. in the case of noisy cross-talk to measure the effective
+    action of some "rotation program" that acts on completely different qubits but nonetheless
+    rotates each measure_qubit.
+
+    The single qubit algorithm is due to:
 
     [RPE]  Robust Calibration of a Universal Single-Qubit Gate-Set via Robust Phase Estimation
             Kimmel et al., Phys. Rev. A 92, 062315 (2015)
             https://doi.org/10.1103/PhysRevA.92.062315
+            arXiv:1502.02677
 
     [RPE2] Experimental Demonstration of a Cheap and Accurate Phase Estimation
             Rudinger et al., Phys. Rev. Lett. 118, 190502 (2017)
             https://doi.org/10.1103/PhysRevLett.118.190502
-
-    :param rotation: the program or gate whose angle of rotation is to be estimated
-    :param num_depths: the number of depths in the protocol described in [RPE]. Max depth = 2**(num_depths-1)
-    :param axis: the axis of rotation specified by (theta, phi). Assumed to be the Z axis if none is specified.
-    :param measurement_qubit: Pertinent only to CPHASE experiment. See generate_2q_single_depth_experiment above
-    :param init_one: Pertinent only to CPHASE experiment. See generate_2q_single_depth_experiment above
-    :return:
+            arXiv:1702.01763v1
+
+    :param rotation: the program or gate whose angle of rotation is to be estimated. Note that
+        this program will be run through forest_benchmarking.compilation.basic_compile().
+    :param change_of_basis: a matrix, gate, or program for the unitary change of basis
+        transformation which maps the computational basis into the basis formed by eigenvectors
+        of the rotation. The sign of the estimate will be determined by which computational basis
+        states are mapped to which eigenvectors. Following the right-hand-rule convention,
+        a rotation of RX(phi) for phi>0 about the +X axis should be paired with a change of basis
+        maps |0> --> |+> and |1> --> |-> . This is achieved by the gate RY(pi/2, qubit).
```

Comment:
change_of_basis essentially replaces the "axis of rotation" argument from before. I allowed for it to be specified as a matrix in case the user didn't have a particular qc topology in mind; they could later supply this with a call to add_programs_to_rpe_dataframe(qc, df) which would then perform the translation for them. It could simplify things to require that this is a program, e.g. then program generation would be automatic in all cases (it already is if change_of_basis is a gate or program), but the other benchmarking methods seemed to only require the qc object at runtime so I wanted to match that behavior.